src/assets/TokensController.ts

```diff
@@ -178,6 +187,10 @@ export class TokensController extends BaseController<
     });
   }
 
+  configureProvider(provider: any) {
+    this.ethersProvider = new ethers.providers.Web3Provider(provider);
```

Comment:
@Gudahtt I don't know if this is the pattern we're looking for, but I saw that in other controllers you were pushing away from setters/getters for configuration effects.

Reply:
Not sure I follow what this method is for. Isn't the provider passed in as config?

Reply:
Yes. It should probably be and then per the comments above, reinstantiated on network change?